pages/in/widget/iscn-ar/index.vue

```diff
@@ -0,0 +1,746 @@
+        this.tags = tags;
+        this.url = url;
+        this.type = type;
+        this.license = license;
+        await this.submitISCNTransfer();
+      }
+    },
+    async submitISCNTransfer() {
+      this.transactionStatus = 'restart';
+      try {
+        const from = await this.fetchCurrentCosmosWallet();
+        if (!from) {
+          throw new Error('PLEASE_RELOGIN');
+        }
+        if (!this.isUsingKeplr) {
+          const { cosmosWallet } = this.getUserInfo;
+          const userWallet = cosmosWallet;
+          if (userWallet !== undefined && from !== userWallet) {
+            throw new Error('VALIDATION_FAIL');
+          }
+        }
+        const signer = await this.prepareCosmosTxSigner();
+        const {
+          fingerprints,
+          title,
+          tags,
+          type,
+          license,
+          url,
+        } = this;
+        const txHash = await this.sendISCNSignature({
+          cosmosWallet: from,
+          userId: this.getUserId || '',
+          displayName: this.getUserInfo.displayName || '',
+          fingerprints,
+          name: title,
+          tags,
+          type,
+          license,
+          url,
+          signer,
+          shouldShowTxDialog: false,
+        });
+        if (txHash) await this.postISCNTransaction({ txHash });
+      } catch (error) {
+        this.transactionStatus = 'failed';
+        console.error(error);
+      }
+    },
+    async postISCNTransaction({ txHash, error } = {}) {
+      this.transactionStatus = 'done';
+      if (this.opener || this.redirectUri) {
+        const ISCNTransferInfo = await getISCNTransferInfo(txHash, { blocking: true });
+        const {
+          isFailed, iscnId, iscnVersion, timestamp,
+        } = ISCNTransferInfo;
+        const success = !isFailed;
+        const payload = {};
+        if (txHash) payload.tx_hash = txHash;
+        if (iscnId) payload.iscnId = iscnId;
+        if (iscnVersion) payload.iscnVersion = iscnVersion;
+        if (timestamp) payload.timestamp = timestamp;
+        if (error) payload.error = error;
+        if (success !== undefined) payload.success = success;
+        if (this.opener) {
+          const message = JSON.stringify({
+            action: 'ISCN_SUBMITTED',
+            data: payload,
+          });
+          window.opener.postMessage(message, this.redirectOrigin);
+          const iscnIdString = encodeURIComponent(iscnId);
+          window.location.href = `https://app.like.co/view/${iscnIdString}?layout=popup`;
+        }
+      }
+    },
+    async onClickRedirectToKeplrSign() {
+      const res = await Keplr.initKeplr();
+      if (!res) {
+        throw new Error('FAILED_CONNECT_TO_KEPLR');
+      }
+      this.setDefaultCosmosWalletSource({ source: 'keplr', persistent: false });
+      this.isUsingKeplr = true;
+      // Direct register ISCN flow
+      if (this.actualSendAmount === 0 || this.actualSendAmount === '0') {
+        this.mainStatus = 'directRegisterISCN';
```

Comment:
`directRegisterISCN` is not a very clear name, what does this do? how about just `registerISCN` ?

Reply:
there is already a 'registerISCN' status for indirectRegistering...  but actually can re-use, will add this after the translation commit

Reply:
merged into one section in commit f2ccd5e